### PR TITLE
Add _get_boto_client and STS token management.

### LINKED ISSUE #24: Support STS token in aws_ssm connection plugin.
##### SUMMARY
The current implementation of the aws_ssm connection plugin relies on the exported environment variables, or on a default connection profile being configured on the controller.
An ideal implementation would allow the task caller to pass an STS token, for example in cases where there is a cross-account trust policy and the node is able to retrieve such session token and execute tasks in the target account.
This would also allow a more versatile usage from the API, by dynamically assume the target role STS session and pass it to each invocation.

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
aws_ssm.py connection plugin

##### ADDITIONAL INFORMATION
This is how a task can be called with all the sts parameters:

```yaml
---
- hosts: all
  vars:
    ansible_aws_ssm_region: us-east-1
    bucket_name: helper-bucket-flavioelawi
    ansible_aws_ssm_access_key_id: <THE_ACCESS_KEY_ID>
    ansible_aws_ssm_secret_access_key: <THE_SECRET_KEY>
    ansible_aws_ssm_session_token: <THE_SESSION_TOKEN>
  tasks:
  - name: test stat
    stat:
      path: /etc/foo.conf
    register: file_details
  - debug:
      msg: "file or dir exists"
    when: file_details.stat.exists
```

### PR DESCRIPTION
**Pull request by flavioelawi**
##### SUMMARY

Fixes #24 

Add the following parameters to aws_ssm.py connection plugin:
* ansible_aws_ssm_access_key_id
* ansible_aws_ssm_secret_access_key
* ansible_aws_ssm_session_token
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

This implements some basic STS token management, that can be passed as parameters to the task when the aws_ssm connection plugin is involved, the parameters are scoped to the plugin namespace.

##### ADDITIONAL INFORMATION

If you have a role that you allowed to assume in a target account, you would need to assume such role in the target account before invoking the task; this example comes from an invocation by using ansible APIs, where even assuming the role in the target account, the connection plugin was still executing under the controller-node account session.
```paste below
flavioel/.virtualenvs/ansible_ssm/lib/python3.6/site-packages/botocore/client.py\", line 626, in _make_api_call\n    raise error_class(parsed_response, operation_name)\nbotocore.errorfactor
y.TargetNotConnected: An error occurred (TargetNotConnected) when calling the StartSession operation: i-01234567890123456 is not connected.\n",
```
By implementing the changes in this PR I am able to pass an STS token as a parameter to the task, letting it execute under the target account context.

This change also allows for backward compatibility as if nothing is specified (no OS environment variables, no parameters) the boto3 client will automatically select the `default` profile configured on the controller.

### CODE REVIEW

**Comment by flavioelawi** [comment]
Pinging the the plugin authors: @psharkey @hanumantharaomvl @gau1991 :)

**Comment by MrBones757** [comment]
Is there any update on when we might expect this change to be merged into stable?
It was originally discussed here: https://github.com/ansible/ansible/pull/49652#issuecomment-569858069
but it was never included in the final release.

Thanks for the PR though - saved me some time as i've just come across this issue myself!

**Review by gau1991 — APPROVED** [review]

**Review by tremble — APPROVED** [review]
Change looks sound, after a little digging my previous comments don't really hold true.  @gau1991 (one of the authors) has +1ed so I think we're good to merge this.

File: plugins/connection/aws_ssm.py [hunk 1]
```diff
@@ -20,6 +20,18 @@
...
+    vars:
+    - name: ansible_aws_ssm_access_key_id
+  secret_access_key:
+    description: The STS secret key to use when connecting via session-manager.
+    vars:
+    - name: ansible_aws_ssm_secret_access_key
+  session_token:
+    description: The STS session token to use when connecting via session-manager.
+    vars:
+    - name: ansible_aws_ssm_session_token
```
Comment (tremble) [inline plugins/connection/aws_ssm.py:34]:
```suggestion
  access_key_id:
    description: The STS access key to use when connecting via session-manager.
    vars:
    - name: ansible_aws_ssm_access_key_id
    version_added: 1.3.0
  secret_access_key:
    description: The STS secret key to use when connecting via session-manager.
    vars:
    - name: ansible_aws_ssm_secret_access_key
    version_added: 1.3.0
  session_token:
    description: The STS session token to use when connecting via session-manager.
    vars:
    - name: ansible_aws_ssm_session_token
    version_added: 1.3.0
```

**Review by tremble — CHANGES_REQUESTED** [review]
Actually, two minor documentation things:
1. Please add a changelog fragment https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to
2. We need a `version_added` entry on the new options.

**Comment by flavioelawi** [comment]
Cool, i'll squash the commits, add the changelog fragment and update the pr, thanks.

**Comment by flavioelawi** [comment]
I made a mess with the pull requests, the correct one is https://github.com/ansible-collections/community.aws/pull/218

sorry about that.

**Pull request closed without merging**